### PR TITLE
Openapi new auth

### PR DESCRIPTION
**Pull request by vlavorini**
## Checklist
- [X] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [X] Docs related to the changes have been added/updated (if applicable)

## Context
In order to interrogate the endpoints, an authentication via token is required.

Before this PR, only a POST request with username/password in the body was available. Now also a GET method, with username/password directly wrote in the url, is available.

### CODE REVIEW

**Comment by xiphl** [comment]
Hi @vlavorini , may I ask if there are any plans to expand the connector to return information about POST endpoints as well?
Currently I only see GET endpoints when I ingest openapi sources.

**Comment by vlavorini** [comment]
Hi @xiphl, in my idea you can obtain datasets only with 'get' methods, but i acknowledge that also POST requests can return data.

There are two theoretical issues with POST:

* it is not [safe](https://developer.mozilla.org/en-US/docs/Glossary/Safe/HTTP)
* it is not [idempotent](https://developer.mozilla.org/en-US/docs/Glossary/Idempotent) 

From the practical point of view, even if you are sure about your endpoints, you need to fill the request Body, and that mean that you have to know what to insert in there.

I'm pretty sure that even faced to the same set of APIs, different POST endpoints need different content in their body. I don't know how to model this.

**Comment by xiphl** [comment]
I see... So these were the considerations behind it. Thanks for sharing!

**Comment by RyanHolstien** [comment]
I think we should make note that using user/pass in the URL should be done cautiously. Depending on set-up, this could log the username & password combo in the webserver logs as a part of the request URL. Otherwise LGTM cc: @dexter-mh-lee

**Review by RyanHolstien — APPROVED** [review]

**Comment by vlavorini** [comment]
Hi @RyanHolstien , do you think it will be good to add few lines in the docs? Cheers

**Comment by RyanHolstien** [comment]
> Hi @RyanHolstien , do you think it will be good to add few lines in the docs? Cheers

Yep! I think that's sufficient

**Comment by RyanHolstien** [comment]
After discussing with the team, we think that this should be done through an auth token rather than directly supplying the username and password. This adds an extra step of reaching out to the auth provider separately from the script to get the auth token, but we believe it to be a more secure pattern to follow.

**Comment by vlavorini** [comment]
I understand, and agree with you. The problem is that there are applications that leave you no choice, like the one I faced, for which I had to implement such a poor method.

**Review by RyanHolstien — APPROVED** [review]

**Comment by RyanHolstien** [comment]
Yeah in the case that your auth provider does not provide an option to produce a token, I think this is the only option that makes sense.

We should still add some docs on the drawbacks of this approach and if possible include another option to support a token in addition to the user/pass combo.

File: metadata-ingestion/src/datahub/ingestion/source/openapi.py [hunk 1]
```diff
@@ -269,7 +297,7 @@ def close(self):
 
 class OpenApiSource(APISource):
     def __init__(self, config: OpenApiConfig, ctx: PipelineContext):
-        super().__init__(config, ctx, "OpenApi")
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/ingestion/source/openapi.py:272]: 1. Is this intentional?
2. what happens to previously ingested urns? when you re-run ingestion they get new urns, so the old entities will continue to exist?

Comment (vlavorini) [reply metadata-ingestion/src/datahub/ingestion/source/openapi.py:272]: Yes, it's intentional. I spotted that there was issues with finding the icon, so I uniformed all to the lowercase format, as in other ingestion plugins.
The older urns will continue to exist as different entities.

**Review by shirshanka — APPROVED** [review]
LGTM

**Pull request merged**